diff --git a/include/Object3d.hpp b/include/Object3d.hpp
index 8aee693..b64dbcb 100644
--- a/include/Object3d.hpp
+++ b/include/Object3d.hpp
@@ -499,11 +499,11 @@ public:
 
 	constexpr const Vector3& operator[](int index) const { return rows[index]; }
 
-	constexpr bool operator==(const Matrix3& other) {
+	constexpr bool operator==(const Matrix3& other) const {
 		return rows[0] == other[0] && rows[1] == other[1] && rows[2] == other[2];
 	}
 
-	constexpr bool IsIdentity() { return *this == Matrix3(); }
+	constexpr bool IsIdentity() const { return *this == Matrix3(); }
 
 	Matrix3& Identity() {
 		//1.0f, 0.0f, 0.0f
@@ -703,9 +703,9 @@ public:
 	constexpr float& operator[](int index) { return m[index]; }
 	constexpr float operator[](int index) const { return m[index]; }
 
-	bool operator==(const Matrix4& other) { return (std::equal(m, m + sizeof m / sizeof *m, other.m)); }
+	bool operator==(const Matrix4& other) const { return (std::equal(m, m + sizeof m / sizeof *m, other.m)); }
 
-	bool IsIdentity() { return *this == Matrix4(); }
+	bool IsIdentity() const { return *this == Matrix4(); }
 
 	Matrix4& Identity() {
 		std::memset(m, 0, sizeof(float) * 16);
